File: backend/core-api/src/main/java/ai/govbiz/core/service/AiServiceHealthService.java
package ai.govbiz.core.service;

import ai.govbiz.core.client.ai.AiServiceClient;
import ai.govbiz.core.client.ai.AiServiceClientException;
import ai.govbiz.core.client.ai.AiServiceHealthPayload;
import org.springframework.stereotype.Service;

@Service
public class AiServiceHealthService {

    private static final String EXPECTED_STATUS = "up";
    private static final String EXPECTED_SERVICE = "govbiz-ai-service";

    private final AiServiceClient aiServiceClient;

    public AiServiceHealthService(AiServiceClient aiServiceClient) {
        this.aiServiceClient = aiServiceClient;
    }

    public AiServiceHealthResult getHealth() {
        AiServiceHealthPayload payload;
        try {
            payload = aiServiceClient.getHealth();
        } catch (AiServiceClientException exception) {
            throw AiServiceHealthException.fromClient(exception);
        }

        if (payload == null
                || !EXPECTED_STATUS.equals(payload.status())
                || !EXPECTED_SERVICE.equals(payload.service())) {
            throw AiServiceHealthException.invalidContract();
        }

        return new AiServiceHealthResult(payload.status(), payload.service());
    }
}

File: backend/core-api/src/main/kotlin/ai/govbiz/core/service/AiServiceHealthService.kt
package ai.govbiz.core.service

import ai.govbiz.core.client.ai.AiServiceClient
import ai.govbiz.core.client.ai.AiServiceClientException
import org.springframework.stereotype.Service

@Service
class AiServiceHealthService(
    private val aiServiceClient: AiServiceClient,
) {
    fun getHealth(): AiServiceHealthResult {
        val payload = try {
            aiServiceClient.getHealth()
        } catch (exception: AiServiceClientException) {
            throw AiServiceHealthException.fromClient(exception)
        }

        val status = payload.status
        val service = payload.service
        if (status != EXPECTED_STATUS || service != EXPECTED_SERVICE) {
            throw AiServiceHealthException.invalidContract()
        }

        return AiServiceHealthResult(status, service)
    }

    private companion object {
        const val EXPECTED_STATUS = "up"
        const val EXPECTED_SERVICE = "govbiz-ai-service"
    }
}
